src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -94,21 +98,18 @@ def get(self, request: Request, project) -> Response:
         )
         with configure_scope() as scope:
             for config in configs:
+                if not filepath.startswith(config.stack_root) and not frame:
+                    latest_error = "stack_root_mismatch"
+                    continue
+
                 result["config"] = serialize(config, request.user)
```

Comment:
I moved the `if` block above so we *only* include a code mapping when there's a match rather than when there are no matches.